api/apps/api/src/migrations/api/1626458359147-AddExternalUniqueIdToApiEvents.ts

```diff
@@ -0,0 +1,20 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddExternalUniqueIdToApiEvents1626458359147
+  implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`ALTER TABLE api_events
+        ADD external_id character varying`);
+
+    await queryRunner.query(
+      `ALTER TABLE api_events ADD CONSTRAINT api_events_external_id_unique UNIQUE ("external_id")`,
```

Comment:
As far as I understand, it relates to some kind of `resource id`.

Why would it need to be unique? Doesn't it mean, you cannot operate on given resource multiple times? (for example, some related PU) Or you cannot include the `external id` within all sub-types of related events? (i.e. use the id for all of `started`, `finished`, `errored` etc)

Reply:
Oh, and adding `+kind` seems strange and potentially causing issues, as it inserts something else than you except to.

Reply:
it relates to an external id of the event

the resource is already identified by `topic`